Migrate Habitica actions to dedicated action pages (#46268)
home-assistant/home-assistant.github.io: source/_actions/habitica.create_habit.markdown

---
title: "Create habit"
action: habitica.create_habit
domain: habitica
description: "Creates a habit for your Habitica character."
---

Use this action to create a habit for the selected Habitica character.

{% include actions/ui_header.md %}

To create a habit from an automation or a script:

1. Go to {% my automations title="**Settings** > **Automations & scenes**" %}.
2. Open an existing automation or script, or select **Create automation** > **Create new automation**.
3. If you're setting up a new automation, add a trigger in the **When** section. Scripts don't need a trigger. They run when something else calls them.
4. In the **Then do** section, select **Add action**.
5. From the search box, search for and select **Habitica: Create habit**.

> 6. Select the **Config entry** of the character and enter a **Name**. Optionally, set the other values.

Reviewer: ```suggestion
6. Select the **Config entry** of the character and enter a **Task name**. Optionally, set the other values.
```

7. Select **Save**.

This action does not support targets. In the UI, you are not prompted to choose an area, device, entity, or label.

### Options in the UI

{% options_ui %}
Config entry:
  description: The Habitica character to create the habit for.
  required: true

> Name:

Reviewer: ```suggestion
Select character:
  description: The Habitica character to create the habit for.
  required: true
Task name:
```

  description: The title for the habit.
  required: true
Notes:
  description: The notes for the habit.
  required: false
Up_down:

>   description: "Defines if the habit is positive, negative, or both. One or more of up and down."

Reviewer: ```suggestion
Rewards or losses:
  description: "Whether the habit is good and rewarding (positive), bad and penalizing (negative), or both."
```

  required: false

> Priority:

Reviewer: ```suggestion
Difficulty:
```

  description: "The difficulty of the habit. One of trivial, easy, medium, or hard."
  required: false

> Frequency:

Reviewer: ```suggestion
Counter reset:
```

  description: "Defines when the habit's counter resets. One of daily, weekly, or monthly."
  required: false
Tag:
  description: Add tags to the habit. If a tag does not exist yet, it is created.
  required: false

> Alias:

Reviewer: ```suggestion
Tags:
  description: Add tags to the habit. If a tag does not exist yet, it is created.
  required: false
Task alias:
```
